src/Congo.WebApi/Controllers/ProductsController.cs

```diff
@@ -28,12 +29,29 @@ public async Task<IEnumerable<Product>> Get()
             return await _mediator.Send(new GetProductListQuery());
         }
 
+
+        [HttpPost]
+        [Route("~/api/{seller}/products")]
+        public ActionResult<Product> Create(ProductInputModel product)
+        {
+            if (!ModelState.IsValid)
+            {
+                return ValidationProblem(ModelState);
+            }
+
+            var productResult = _mediator.Send(new InsertProductCommand
```

Comment:
Use await here
[SO Thread](https://stackoverflow.com/questions/32239661/await-vs-task-result-in-an-async-method)